plugin/restore/mysql/mysql.go

```diff
@@ -50,9 +54,64 @@ func New(driver *mysql.Driver, instance *mysqlutil.Instance, connCfg db.Connecti
 	}
 }
 
-// replayBinlog restores the database using incremental backup in time range of [config.Start, config.End).
-func replayBinlog(ctx context.Context, config api.BinlogInfo, binlogDir string) error {
-	return fmt.Errorf("Unimplemented")
+// ReplayBinlog replays the binlog about `originDatabase` from `startBinlogInfo.Position` to `targetTs`.
+func (r *Restore) replayBinlog(ctx context.Context, originDatabase, pitrDatabase, binlogDir string, startBinlogInfo api.BinlogInfo, targetTs int64) error {
+	if err := r.SyncLatestBinlog(ctx, binlogDir); err != nil {
+		return err
+	}
+
+	replayBinlogPaths, err := r.getReplayBinlogPathList(ctx, startBinlogInfo, binlogDir)
+	if err != nil {
+		return err
+	}
+
+	stopTime := time.Unix(targetTs, 0)
+	mysqlbinlogArgs := []string{
+		"--disable-log-bin",
+		fmt.Sprintf(`--rewrite-db=%s->%s`, originDatabase, pitrDatabase),
+		fmt.Sprintf("--database=%s", pitrDatabase),
+		fmt.Sprintf("--start-position=%d", startBinlogInfo.Position),
+		fmt.Sprintf(`--stop-datetime=%d-%d-%d %d:%d:%d`, stopTime.Year(), stopTime.Month(), stopTime.Day(), stopTime.Hour(), stopTime.Minute(), stopTime.Second()),
+	}
+
+	mysqlbinlogArgs = append(mysqlbinlogArgs, replayBinlogPaths...)
+
+	mysqlArgs := []string{
+		fmt.Sprintf("--host=%s", r.connCfg.Host),
+		fmt.Sprintf("--port=%s", r.connCfg.Port),
+		fmt.Sprintf("--user=%s", r.connCfg.Username),
+		fmt.Sprintf("--password=%s", r.connCfg.Password),
+	}
+
+	mysqlbinlogCmd := exec.CommandContext(ctx, r.mysqlutil.GetPath(mysqlutil.MySQLBinlog), mysqlbinlogArgs...)
+	mysqlCmd := exec.CommandContext(ctx, r.mysqlutil.GetPath(mysqlutil.MySQL), mysqlArgs...)
+
+	mysqlRead, mysqlbinlogWrite := io.Pipe()
```

Comment:
Please use `mysqlbinlogCmd.StdoutPipe()`. It's cleaner.

ref: https://stackoverflow.com/a/53351345/2497338